File: CartMinder/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/cartminder/dao/ItemDAO.java
package id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.dao;

import androidx.annotation.WorkerThread;
import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder.model.Item;

@Dao
public interface ItemDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Item item);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void update(Item item);

    @Delete
    void delete(Item item);

    @Query("SELECT * FROM item_table")
    List<Item> getItems();

    @Query("SELECT * FROM item_table WHERE cartId=:cartId")
    List<Item> getCartItems(long cartId);
}

File: CartMinder2/app/src/main/java/id/ac/ui/cs/mobileprogramming/kevinlh/cartminder2/database/dao/ItemDao.kt
package id.ac.ui.cs.mobileprogramming.kevinlh.cartminder2.database.dao

import androidx.room.*
import id.ac.ui.cs.mobileprogramming.kevinlh.cartminder2.model.Item
import kotlinx.coroutines.flow.Flow


@Dao
interface ItemDao {
    @Insert
    suspend fun insert(item: Item): Long

    @Update
    suspend fun update(item: Item)

    @Delete
    suspend fun delete(item: Item)

    @Query("SELECT * FROM item_table WHERE cartId=:cartId")
    suspend fun findByCartId(cartId: Long): List<Item>

    @Query("SELECT * FROM item_table")
    fun findAll(): Flow<List<Item>>
}